### PR TITLE
fix: signing with ledger throws an empty error

### LINKED ISSUE #2650: Signing with ledger throws an empty error
### Summary of Bug

When attempting to sign any TX with ledger hardware wallet an empty error is thrown with a second delay. 

```
❯ celestia-appd tx sign tx-test.json --from ledger-test   --chain-id="mocha-4" --offline -s 0 -a 0 --ledger
Default sign-mode 'direct' not supported by Ledger, using sign-mode 'amino-json'.
Error:
```

I tried playing around with various options and flags - the result is always the same

### Version

v1.0.0

### Steps to Reproduce

* Have working ledger and ensure its connected and unlocked
* Add ledger key via `celestia-appd keys add <name> --ledger`
  * This prompts perm on ledger and this works fine
* Generate any test tx with `--generate-only`
* Sign transaction
`celestia-appd tx sign tx-test.json --from <name>  --chain-id="mocha-4" --offline -s 0 -a 0 --ledger`

### For Admin Use

- [X] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

Closes #2650 by upgrading to celestiaorg/cosmos-sdk [v1.18.2-sdk-v0.46.14](https://github.com/celestiaorg/cosmos-sdk/releases/tag/v1.18.2-sdk-v0.46.14)

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by cmwaters** [comment]
You don't want to just bump it on the SDK?

Ref: https://github.com/celestiaorg/cosmos-sdk/pull/348

I'm fine with either. This is probably quicker because you don't need a second release

**Comment by rootulp** [comment]
I think we may need to bump it in celestiaorg/cosmos-sdk and then here. Will start that now

```
$ make build && make install
--> Updating go.mod
# github.com/cosmos/cosmos-sdk/crypto/ledger
../../../../go/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.18.1-sdk-v0.46.14/crypto/ledger/ledger_real.go:18:10: cannot use device (variable of type *ledger_cosmos_go.LedgerCosmos) as SECP256K1 value in return statement: *ledger_cosmos_go.LedgerCosmos does not implement SECP256K1 (wrong type for method SignSECP256K1)
		have SignSECP256K1([]uint32, []byte, byte) ([]byte, error)
		want SignSECP256K1([]uint32, []byte) ([]byte, error)
make: *** [build] Error 1
```

File: go.mod [hunk 1]
```diff
@@ -44,6 +44,7 @@ require (
 	github.com/bits-and-blooms/bitset v1.7.0 // indirect
 	github.com/consensys/bavard v0.1.13 // indirect
 	github.com/consensys/gnark-crypto v0.12.0 // indirect
+	github.com/cosmos/ledger-cosmos-go v0.13.1 // indirect
```
Comment (rootulp) [inline go.mod:47]: 👍

**Review by rootulp — DISMISSED** [review]
LGTM if CI passes

**Review by cmwaters — APPROVED** [review]

**Comment by rootulp** [comment]
FYI we can merge this now or wait for a Cosmos SDK v0.46.x release (see https://github.com/celestiaorg/cosmos-sdk/pull/349#issuecomment-1760136839) which is expected next week

**Review by rootulp — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]
Looks good to me! 
[optional] Regarding the PR title, is intended change to introduce the behavior of throwing an empty error, or preventing errors from being thrown? It might be more informative to emphasize the intended change rather than the initial bug (entirely optional though).

**Comment by rootulp** [comment]
Good point. This PR prevents an error from being thrown. I updated the PR description and PR title.

**Pull request merged**